iotdb-core/ainode/ainode/core/manager/inference_manager.py

```diff
@@ -229,18 +353,43 @@ def _run(
                     predict_length,
                 )
 
-            if model_id == self.ACCELERATE_MODEL_ID and self.DEFAULT_POOL_SIZE > 0:
-                # TODO: Logic in this branch shall handle all LTSM inferences
+            if model_id in self.ACCELERATE_MODEL_ID and "cuda" in str(
+                self.DEFAULT_DEVICE
+            ):
+                # lazy initialization for first request
+                if model_id not in self._request_pool_map:
+                    with self._pool_init_lock:
```

Comment:
Currently, this lock seems useless. A useful lock can be set to protect the route and create behaviour for a specific type of request pool